plugins/catalog-backend/src/service/filterQuery.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { InputError } from '@backstage/backend-common';
+import { EntityFilters } from '../database';
+
+export function translateQueryToEntityFilters(
+  query: Record<string, any>,
+): EntityFilters[] {
+  if (!query.filter) {
+    return [];
+  }
+
+  const filterStrings = Array.isArray(query.filter)
```

Comment:
`[query.filter].flat()` would be an option here

Reply:
Aw yiss, I forget